src/SDK/SmartDetectors.Extensions/AlertExtensions.cs

```diff
@@ -124,17 +100,71 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
         }
 
         /// <summary>
-        /// Creates an <see cref="AlertProperty"/> based on an alert presentation property
+        /// Extract all alert properties from the specified object
         /// </summary>
-        /// <param name="alert">The alert</param>
+        /// <param name="propertiesOwner">The object from which to extract the properties</param>
+        /// <param name="order">The order to use</param>
+        /// <returns>The extracted properties</returns>
+        private static List<AlertProperty> ExtractProperties(object propertiesOwner, Order order)
+        {
+            if (propertiesOwner == null)
+            {
+                return new List<AlertProperty>();
+            }
+
+            // Collect all object properties, and sort them by order
+            var orderedProperties = propertiesOwner.GetType().GetProperties().Select(property =>
+                {
+                    // Get the property value
+                    object propertyValue = property.GetValue(propertiesOwner);
+                    string propertyStringValue = PropertyValueToString(propertiesOwner, property, propertyValue);
```

Comment:
>PropertyValueToString [](start = 49, length = 21)

i don't think we need to call this here - it's a weird implementation (i know you just copied it)

please change the below condition to be:
if (propertyValue  == null || 
    (propertyValue is string stringValue && string.IsNullOrWhiteSpace(stringValue)) ||
    (propertyValue is ICollection collectionValue && collectionValue.Count == 0))